### ISSUE TITLE
Marker onTap event is not fired since 6.2.20

### ISSUE THREAD

**Issue by foxgreek** [nativescript-community/ui-mapbox#102]
After updating to 6.2.20, the onTap event of markers on the map is not fired anymore. Going down to 6.2.19 solved the problem.

**Comment by farfromrefug** [comment]
@mohammadrafigh seems your change broke other things. Can you look at it?

**Comment by paulmlilo** [comment]
I experienced the same thing as well.

**Comment by mohammadrafigh** [comment]
@farfromrefug Well, It seems we need to check if there is an onTap listener available then we should bypass default listener. But if there is no onTap listener, then we can allow custom tap listeners. This way there is no possibility to have both. I will take a look to see if we can have both behaviors at the same time. So for now the solution for who uses onTap listener is to use 6.2.19. I will send a PR soon, hope this time it fixes everything and gets merged :)).

**Comment by foxgreek** [comment]
@mohammadrafigh it is still broken in 6.2.27

**Comment by farfromrefug** [comment]
@foxgreek yes this repo is looking for maintainers. I personally. dont use mapbox anymore.
If you want to have a go at it please do

**Comment by mohammadrafigh** [comment]
Hi Guys, I'm not able to reproduce your problem, It's working here. Can you provide a demo project so I can take a look?

**Comment by kakha13** [comment]
It is not working on iOS

**Comment by mohammadrafigh** [comment]
It's working on my projects. I'm not able to fix it unless someone provides a demo project that reproduces the problem. Otherwise I consider that it works and something might be wrong with your usage.

**Comment by kakha13** [comment]
> It's working on my projects. I'm not able to fix it unless someone provides a demo project that reproduces the problem. Otherwise I consider that it works and something might be wrong with your usage.

Hey thanks, here is demo
[mapbox-marker-demo](https://github.com/kakha13/mapbox-marker-demo)

**Comment by mohammadrafigh** [comment]
Good I will check this out after the weekend. Thank you

**Comment by kakha13** [comment]
I compared the current version with 6.2.19 and found this difference.
inside `index.ios.js `function: `setOnMapClickListener`
```js
 if (recognizer instanceof UITapGestureRecognizer) {
      recognizer.addTargetAction(theMap['mapTapHandler'], 'tap');
      break;
  }
```
Change:
```js
if (recognizer instanceof UITapGestureRecognizer) {
    tapGestureRecognizer.requireGestureRecognizerToFail(recognizer);
}
```

Tap on marker working now.
